src/Maestro/Maestro.MergePolicies/ValidateCoherencyMergePolicy.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Maestro.Contracts;
+using Maestro.MergePolicyEvaluation;
+using Microsoft.DotNet.DarcLib;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Maestro.MergePolicies;
+
+public class ValidateCoherencyMergePolicy : MergePolicy
+{
+    public override string DisplayName => "Validate coherency";
+
+    public override Task<MergePolicyEvaluationResult> EvaluateAsync(IPullRequest pr, IRemote darc)
+    {
+        if (pr.CoherencyCheckSuccessful)
+            return Task.FromResult(Succeed("Coherency check successful."));
+
+        StringBuilder description = new StringBuilder("Coherency update failed for the following dependencies:");
+        foreach (CoherencyErrorDetails error in pr.CoherencyErrors)
+        {
+            description.Append("\n * ").Append(error.Error);
```

Comment:
You can use AppendLine instead of Append so you can save yourself adding the \n manually